recipes/perf/all/conanfile.py

```diff
@@ -0,0 +1,49 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches
+import os
+
+required_conan_version = ">=1.35.0"
+
+class Perf(ConanFile):
+    name = "perf"
+    description = "Linux profiling with performance counters"
+    topics = ("linux", "profiling")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://perf.wiki.kernel.org/index.php"
+    license = "GPL-2.0 WITH Linux-syscall-note"
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = "patches/*"
+
+    _source_subfolder = "source_subfolder"
+
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("perf is supported only on Linux")
+
+    def build_requirements(self):
+        self.build_requires("flex/2.6.4")
+        self.build_requires("bison/3.5.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def build(self):
+        apply_conandata_patches(self)
```

Comment:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```


There is a change in Conan 1.40 which made apply_conandata_patches invalid for CCI.